src/precice/impl/ValidationMacros.hpp

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+/**@file
+ * This file contains verification macros to harden the SolverInterface against misuse and misconfiguration.
+ *
+ * \attention Only include this file from SolverInterfaceImpl.cpp
+ */
+
+//
+// MESH VALIDATION 
+//
+
+/// Implementation of PRECICE_VALIDATE_MESH_ID
+#define PRECICE_VALIDATE_MESH_ID_IMPL(id) \
+    CHECK(_dataIDs.find(id) != _dataIDs.end(), "There is no Mesh with ID:" << id);
+
+/// Implementation of PRECICE_REQUIRE_MESH_READ
+#define PRECICE_REQUIRE_MESH_READ_IMPL(id) \
+    PRECICE_VALIDATE_MESH_ID_IMPL(id) \
+    CHECK(_accessor->isMeshUsed(id), "Mesh is not used by this participant! ID:" << id);
+
+/// Implementation of PRECICE_REQUIRE_MESH_WRITE
+#define PRECICE_REQUIRE_MESH_WRITE_IMPL(id) \
+    PRECICE_REQUIRE_MESH_READ_IMPL(id) \
+    MeshContext& context = _accessor->meshContext(id); \
+    CHECK(context.provideMesh, "Can only write to provided meshes! " << id);
```

Comment:
I don't like the "write" here too much. What about: "Mesh elements (vertices, edges, triangles, quads) can only be created for provided meshes." ?